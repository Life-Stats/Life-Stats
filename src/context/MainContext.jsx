import { createContext, useContext, useState } from 'react';

const MainContext = createContext();

const MainProvider = ({ children }) => {
  const [name, setName] = useState('');

  return (
    <MainContext.Provider value={{ name, setName }}>
      {children}
    </MainContext.Provider>
  );
};

const useMain = () => {
  const context = useContext(MainContext);

  if (context === undefined) {
    throw new Error('useMain must be used within a MainContext Provider');
  }
  return context;
};

export { MainProvider, useMain };