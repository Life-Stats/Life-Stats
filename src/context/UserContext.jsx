import { createContext, useContext, useMemo, useState } from 'react';
import { getUser } from '../services/users';

const UserContext = createContext();

const UserProvider = ({ children }) => {
  const currentUser = getUser();
  // using the lazy initializer method would be good here
  const [user, setUser] = useState(() => {
    return currentUser ? { id: currentUser.id, email: currentUser.email } : {}
  }
);

  const value = useMemo(() => ({ user, setUser }), [user]);

  return <UserContext.Provider value={value}>{children}</UserContext.Provider>;
};

const useUser = () => {
  const context = useContext(UserContext);

  if (context === undefined) {
    throw new Error('useUser must be used within a UserProvider');
  }

  return context;
};

// Dont need to export the context since we use it in the useUser() hook
export { UserProvider, useUser };