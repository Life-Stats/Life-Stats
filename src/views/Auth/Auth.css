.authSect {

}

.authWelcome {
    text-align: center;
    font-size: 30px;

}

.btnDiv {
    text-align: center;
    
}

.imgDiv {
    background-color: white;
    padding: 0px;
}

.btnSect {
    display: inline-flex;
    justify-content: space-between;
}

.btnP {
    margin-left: 5px;
}

.authBtn {
    background-color: rgba(72, 152, 238, 0.662);
    border-radius: 3%;
    padding: 0px;
    margin: 0px;
    box-shadow: 1px 2px grey;
}

.googleBtn {
    padding: 2px;
    background-color: white;
    margin-top: 5px;
    margin-right: 5px;
    height: 30px;
    width: 30px;
}