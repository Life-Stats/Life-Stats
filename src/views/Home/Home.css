body {
  background-color: rgb(254, 249, 239);
  margin: 0%;
  -webkit-background-size: cover;
  -moz-background-size: cover;
  -o-background-size: cover;
  background-size: cover;
  background-repeat: no-repeat;
}

.div {
  background-color: rgba(254, 109, 115);
  border-radius: 1rem;
  text-align: center;
  margin-left: 5%;
  margin-right: 5%;
  box-shadow: black 1px 1px 6px 1px;
}

h1 {
  font-size: 3em;
  margin: 0;
}

.titleDiv {
  text-align: center;
  margin-top: 20px;
}

.desc {
  font-size: 30px;
}

.section {
  background-color: rgba(254, 249, 239);
  border: 0.5em solid rgba(254, 109, 115);
  margin: 2%;
}

.btnDiv {
  margin: 10% 30%;
}

.loading {
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: 0 auto;
  margin-top: 200px;
  font-size: xx-large;
  width: 100%;
  height: 100px;
  color: rgba(254, 109, 115);
  flex-wrap: nowrap;
  align-content: center;
  align-items: center;
  animation: growIcon 3s infinite;
}
.icon {
  border: none;
  box-shadow: none;
}

.signin {
  appearance: button;
  background-color: #1899d6;
  border: solid transparent;
  border-radius: 16px;
  border-width: 0 0 4px;
  box-sizing: border-box;
  color: #ffffff;
  cursor: pointer;
  display: inline-block;
  font-family: din-round, sans-serif;
  font-size: 15px;
  font-weight: 700;
  letter-spacing: 0.8px;
  line-height: 20px;
  margin: 0;
  outline: none;
  overflow: visible;
  padding: 13px 16px;
  text-align: center;
  text-transform: uppercase;
  touch-action: manipulation;
  transform: translateZ(0);
  transition: filter 0.2s;
  user-select: none;
  -webkit-user-select: none;
  vertical-align: middle;
  white-space: nowrap;
  width: 100%;
}

.signin:after {
  background-clip: padding-box;
  background-color: #f61c77dc;
  border: solid transparent;
  border-radius: 16px;
  border-width: 0 0 4px;
  bottom: -4px;
  content: '';
  left: 0;
  position: absolute;
  right: 0;
  top: 0;
  z-index: -1;
}

.signin:main,
.singin:focus {
  user-select: auto;
}

.singin:hover:not(:disabled) {
  filter: brightness(1.1);
}

.signin:disabled {
  cursor: auto;
}
