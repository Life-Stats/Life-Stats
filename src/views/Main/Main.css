.formSubmit {
  text-align: center;
}

.infoSection {
  text-align: center;
}

.infoSection p {
  display: flex;
  justify-content: center;
  margin: 400px 0 800px 0;
  font-size: 2em;
}

.divSize {
  border: 1px black solid;
}

form {
  display: inline-block;
  display: flex;
  flex-direction: column;
  justify-content: center;
  margin: 0 auto;
  width: 300px;
  margin-top: 400px;
  padding: 20px;
  font-size: 2em;
  width: 450px;
  border: 4px rgba(254, 109, 115) solid;
  box-shadow: black 1px 1px 6px 1px ;
}

form label {
  margin-bottom: 20px;
  font-weight: bold;
  font-size: 40px;
}

form button {
  width: 100px;
  display: flex;
  justify-content: center;
  margin: 0 auto;
  margin-top: 20px;
}

.pTagAnimations {
  border: 2px solid black;
  height: 200px;
  overflow: hidden;
  /* background-color: rgb(214, 230, 214); */
}

* {
  box-sizing: border-box;
}
.face {
  display: flex;
  justify-content: space-between;
  width: 150px;
  height: 50px;
  padding: 30px;
  border-top-left-radius: 50%;
  border-top-right-radius: 50%;
  background: tomato;
  transition: 0.3s ease-in-out;
}
.face:hover {
  background: rgb(34, 207, 238);
}

.blinks {
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
}

.breathface {
  display: flex;
  flex-direction: row;
  flex-wrap: wrap;
  justify-content: space-between;
  width: 150px;
  height: 50px;
  padding: 30px;
  background: tomato;
  transition: 0.3s ease-in-out;
}
.breathbottomface {
  display: flex;
  flex-direction: row;
  flex-wrap: nowrap;
  justify-content: center;
  align-items: flex-end;
  width: 150px;
  height: 50px;
  padding: 30px;
  background: tomato;
  transition: 0.3s ease-in-out;
}
.breathface:hover {
  background: rgb(34, 207, 238);
}
.eye {
  background: white;
  width: 25px;
  height: 25px;
  margin: 0 5px 0 5px;
  border-radius: 50%;
  transform-origin: 50%;
  animation: blink 3s infinite;
}
.mouth {
  background: rgb(53, 43, 43);
  width: 20px;
  height: 20px;
  border-radius: 50%;
  margin-left: 2%;
  transform-origin: 50%;
  animation: breaths 2s infinite;
}

.breathbox {
  display: flex;
  justify-content: center;
  flex-wrap: wrap;
  margin-top: 400px
}

.heart {
  background-color: red;
  display: inline-block;
  height: 80px;
  margin: 0 10px;
  position: relative;
  top: 0;
  transform: rotate(-45deg);
  width: 80px;
}

.heart:before,
.heart:after {
  content: '';
  background-color: red;
  border-radius: 50%;
  height: 80px;
  position: absolute;
  width: 80px;
}

.heart:before {
  top: -33px;
  left: 0;
  animation: heartbeats 1s infinite;
}

.heart:after {
  left: 33px;
  top: 0;
  animation: heartbeats 1s infinite;
}

.dream {
  width: 400px;
  height: 200px;
}

.extinct {
  width: 400px;
}

.asleep {
  width: 250px;
  height: 220px;
}

.born {
  margin-top: 400px;
}

@keyframes blink {
  0%,
  100% {
    transform: scale(1, 0.05);
  }
  5%,
  95% {
    transform: scale(1, 1);
  }
}

@keyframes breaths {
  0% {
    transform: scale(1, 1);
  }
  25% {
    transform: scale(1.2, 1.2);
  }
  50% {
    transform: scale(1.5, 1.5);
  }
  75% {
    transform: scale(2, 2);
  }
  90% {
    transform: scale(1.2, 1.2);
  }
}

@keyframes heartbeats {
  25% {
    transform: scale(1.01, 1.01);
  }
  50% {
    transform: scale(1.1, 1.1);
  }
}
