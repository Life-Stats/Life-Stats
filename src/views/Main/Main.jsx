import React, { useState, useEffect } from 'react';
import 'aos/dist/aos.css'; // You can also use <link> for styles
import Styles from './Main.css';
import Aos from 'aos';
import { getHolidays, getMainData } from '../../utils/Utils.jsx';
import tas from '../../assets/tas-removebg-preview.png';
import dreamingPic from '../../assets/dreamPic.png';
import longHair from '../../assets/longHair.png';
import sleepy from '../../assets/sleepy.png';
import born from '../../assets/born.png';

export default function Main() {
  const [date, setDate] = useState('');
  const [breathsState, setBreathsState] = useState('');
  const [heartBeats, setHeartBeats] = useState('');
  const [blinks, setBlinks] = useState(0);
  const [yearsAsleep, setYearsAsleep] = useState('');
  const [dreamDays, setDreamDays] = useState('');
  const [globalExtinction, setGlobalExtinction] = useState('');
  const [show, setShow] = useState(true);
  const [holiday, setHoliday] = useState();
  const [hairGrowth, setHairGrowth] = useState('');
  const [numDays, setNumDays] = useState('');
  const [loading, setLoading] = useState('false');

  useEffect(() => {
    Aos.init({ duration: 3000 });
  }, []);

  function ticker() {
    setInterval(() => {
      setBlinks((prevState) => prevState + 1);
    }, 3000);
    setInterval(() => {
      setHeartBeats((prevState) => prevState + 1);
    }, 900);
    setInterval(() => {
      setBreathsState((prevState) => prevState + 1);
    }, 3000);
  }

  const handleSubmit = async (event) => {
    event.preventDefault();

    ticker();
    setShow(false);
    setLoading(true);
    setHoliday(await getHolidays(date));
    const allObjectsData = await getMainData(date);

    const totalNumDays = allObjectsData.totalNumOfDays;
    setNumDays(totalNumDays);

    const breathTotal = allObjectsData.newTotalBreaths;
    setBreathsState(breathTotal);

    const finalHeartBeats = allObjectsData.totalHeartBeats;
    setHeartBeats(finalHeartBeats);

    const finalYearsAsleep = allObjectsData.totalYearsAsleep;
    setYearsAsleep(finalYearsAsleep);

    const finalBlinks = allObjectsData.totalBlinks;
    setBlinks(finalBlinks);

    const finalDreamDays = allObjectsData.totalDreamDays;
    setDreamDays(finalDreamDays);

    const finalExtinctRate = allObjectsData.totalExtinct;
    setGlobalExtinction(finalExtinctRate);

    const finalHairGrowth = allObjectsData.totalHairGrowth;
    setHairGrowth(finalHairGrowth);
    if (finalHairGrowth !== null) {
      setTimeout(() => setLoading(false), 3000);
    }
  };

  return (
    <>
      {show && (
        <div className={Styles.formSubmit}>
          <form onSubmit={handleSubmit} className="Main">
            <label htmlFor="start">Enter your Birthday </label>
            <input
              id="start"
              type="date"
              value={date}
              onChange={(event) => setDate(event.target.value)}
            />
            <button aria-label="button" type="submit" value="submit">
              Submit
            </button>
          </form>
        </div>
      )}
      {!show && loading && (
        <section data-aos="fade-in" className={Styles.loading}>
          <img
            className={Styles.icon}
            src="https://freesvg.org/img/1539121879.png"
          />
          Loading...
        </section>
      )}
      {!loading && (
        <section data-aos="fade-in" className={Styles.infoSection}>
          <div>
            {numDays ? (
              <>
                {<img src={born} className={Styles.born}></img>}
                <p aria-label="holiday" data-aos="fade-up">
                  A lot has happened in the {`${numDays.toLocaleString()}`} days
                  since you were born.
                </p>
              </>
            ) : null}
          </div>

          <div>
            {numDays ? (
              <p aria-label="holiday" data-aos="fade-up">
                Since you were born...
              </p>
            ) : null}
          </div>

          <div className={Styles.divWrapper}>
            {breathsState ? (
              <>
                <section className={Styles.breathbox}>
                  <div>
                    <div className={Styles.breathface}>
                      <div className={Styles.eye} />
                      <div className={Styles.eye} />
                    </div>
                    <div className={Styles.breathbottomface}>
                      <div className={Styles.mouth} />
                    </div>
                  </div>
                </section>

                <p data-aos="fade-left">
                  You've taken {`${breathsState.toLocaleString()}`} breaths!
                </p>
              </>
            ) : null}
          </div>

          <div>
            {heartBeats ? (
              <>
                <div className={Styles.heart}></div>
                <p data-aos="fade-right">
                  You're heart has beaten {`${heartBeats.toLocaleString()}`}{' '}
                  times!
                </p>
              </>
            ) : null}
          </div>

          <div>
            {blinks ? (
              <>
                <div className={Styles.blinks}>
                  <div className={Styles.face}>
                    <div className={Styles.eye} />
                    <div className={Styles.eye} />
                  </div>
                </div>
                <p data-aos="fade-up">
                  You have blinked {`${blinks.toLocaleString()}`} times!
                </p>
              </>
            ) : null}
          </div>

          <div>
            {yearsAsleep ? (
              <>
                {<img src={sleepy} className={Styles.asleep}></img>}
                <p data-aos="fade-up">
                  {`${yearsAsleep}`} years of your life has been spent asleep!
                </p>
              </>
            ) : null}
          </div>

          <div>
            {dreamDays ? (
              <>
                {<img src={dreamingPic} className={Styles.dream}></img>}
                <p data-aos="fade-up">
                  {`${dreamDays.toLocaleString()}`} days of your life has been
                  spent just dreaming!
                </p>
              </>
            ) : null}
          </div>

          <div>
            {globalExtinction ? (
              <>
                {<img src={tas} className={Styles.extinct}></img>}
                <p data-aos="fade-up">
                  Some estimated up to {`${globalExtinction.toLocaleString()}`}{' '}
                  species of animal life have since gone extinct..
                </p>
              </>
            ) : null}
          </div>

          <div>
            {hairGrowth ? (
              <>
                {<img src={longHair} className={Styles.hair}></img>}
                <p data-aos="fade-up">
                  Your hair has grown a grand total of {`${hairGrowth}`} feet
                  since you were born!
                </p>
              </>
            ) : null}
          </div>

          <div>
            {holiday ? (
              <p aria-label="holiday" data-aos="fade-up">
                Did you know {`${holiday}`} lands on your birthday!?
              </p>
            ) : null}
          </div>

          <div>
            {numDays ? (
              <p aria-label="holiday" data-aos="fade-up">
                .. A lot has happened since you were born, and you still have a
                long way to go.
              </p>
            ) : null}
          </div>

          <div>
            {numDays ? (
              <p aria-label="holiday" data-aos="fade-up">
                Who else knows what else will happen in your lifetime!
              </p>
            ) : null}
          </div>

          <div>
            {numDays ? (
              <p aria-label="holiday" data-aos="fade-up">
                Thanks for playing Life Stats with us :)
              </p>
            ) : null}
          </div>
        </section>
      )}
    </>
  );
}
