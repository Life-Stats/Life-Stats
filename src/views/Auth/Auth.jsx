
import { useHistory } from 'react-router-dom';
import { useUser } from '../../context/UserContext';
import { signInUser, signInUserGitHub } from '../../services/users';
import Styles from './Auth.css'
// cal fetch utils function to gain access to user data from back end response

export default function Auth({ isSigningIn = false }) {
  const history = useHistory();
  const { setUser } = useUser();

  const handleSubmit = async () => {
    try {
      //services need to put in BE and make call to talk to Supabase
      // if they are signing up
        const user = await signInUser();
        setUser(user);
        console.log(user);
        // if (user) return <Redirect to="/main" />;

        console.log('user', user);
    } catch (error) {
      throw error;
    }
  };

  const handleSubmitGitHub = async () => {
    try {
      //services need to put in BE and make call to talk to Supabase
      // if they are signing up
        const user = await signInUserGitHub();
        setUser(user);
        console.log('user', user);
    } catch (error) {
      throw error;
    }
  };


  return (
    <>
   
    <div className={Styles.btnDiv}>
      <section className={Styles.authSect}>
        <h2 className={Styles.authWelcome}>Welcome!</h2>
        <p>Click the button to Sign in with Google</p>
      </section>
      <button className={Styles.authBtn} onClick={handleSubmit}>
        <section className={Styles.btnSect}>
          <div className={Styles.imgDiv}>
            <img  className={Styles.googleBtn} src='https://freesvg.org/img/1534129544.png' />
          </div>
            <p className={Styles.btnP}>Sign in with Google</p>
          </section>
      </button>

      <button className={Styles.authBtn} onClick={handleSubmitGitHub}>
        <section className={Styles.btnSect}>
          <div className={Styles.imgDiv}>
            <img  className={Styles.googleBtn} src='https://upload.wikimedia.org/wikipedia/commons/thumb/3/3f/Github-circle_%28CoreUI_Icons_v1.0.0%29.svg/2048px-Github-circle_%28CoreUI_Icons_v1.0.0%29.svg.png' />
          </div>
            <p className={Styles.btnP}>Sign in with GitHub</p>
          </section>
      </button>
    </div>

    </>
  );
}