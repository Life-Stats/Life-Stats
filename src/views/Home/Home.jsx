import { Link, Redirect } from 'react-router-dom';
import { useUser } from '../../context/UserContext';


export default function Home() {
  const auth = useUser();//make a call to BE for useUser
  if (auth.user.email) return <Redirect to="/main" />;

  return (
    <section>
      <h1>Welcome to Life Stats.... </h1>
      <p>Description of app goes here..</p> 
      <p>
        Please <Link to="/login">sign in</Link>.
      </p>
    </section>
  );
}