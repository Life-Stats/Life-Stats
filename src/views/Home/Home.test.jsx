import React from 'react';
import { render, screen, waitFor } from '@testing-library/react';
import userEvent from '@testing-library/user-event';
import { MemoryRouter, Route, Switch } from 'react-router-dom';
import { UserProvider } from '../../context/UserContext';
import Home from './Home';
import Main from '../Main/Main';
import Auth from '../Auth/Auth';

it('should render the Home view and then redirect to the Main view when clicking the play button ', async () => {
  const { container } = render(
    <UserProvider>
      <MemoryRouter initialEntries={['/']}>
        <Switch>
          <Route exact path="/">
            <Home />
          </Route>
          <Route exact path="/login">
            <Auth />
          </Route>
          <Route exact path="/main">
            <Main />
          </Route>
        </Switch>
      </MemoryRouter>
    </UserProvider>
  );

  expect(container).toMatchSnapshot();
  screen.getByLabelText(/loading/i);

  // return await waitFor(() => {
  //   const button = screen.getByRole('button', { name: 'PLAY' });

  //   expect(button).toBeInTheDocument();

  //   userEvent.click(button);
  //   screen.findByText('Please sign in with Google or GitHub!');
  // });
});
