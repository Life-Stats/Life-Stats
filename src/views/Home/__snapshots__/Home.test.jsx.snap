// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Home view and then redirect to the Main view when clicking the play button  1`] = `
<div>
  <div
    class="titleDiv"
  >
    <h1
      class="title"
    >
      Life Stats
    </h1>
  </div>
  <div
    class="div"
  >
    <section
      class="section"
    >
      <p
        class="desc"
      >
        Welcome! 
        <br />
         We invite you to sign in below before you enjoy the experience.
      </p>
      <div
        class="btnDiv"
      >
        <button
          class="signin"
        >
          Please sign in.
        </button>
      </div>
    </section>
  </div>
</div>
`;
